### PR TITLE
Add POI multiselection

### PR DESCRIPTION
**Pull request by rominf**
Now it's possible to display casino and church locations at the same time!
Ping @vshcherb

### CODE REVIEW

**Comment by rominf** [comment]
Search must be refactored before merging this.

File: OsmAnd/src/net/osmand/plus/activities/MapActivityLayers.java [hunk 1]
```diff
@@ -269,41 +271,51 @@ public AlertDialog selectPOIFilterLayer(final OsmandMapTileView mapView, final P
...
 	}
 
-	private void addFilterToList(final ContextMenuAdapter adapter, final List<PoiUIFilter> list, PoiUIFilter f) {
+	private void addFilterToList(final ContextMenuAdapter adapter, final List<PoiUIFilter> list, final PoiUIFilter f) {
 		list.add(f);
 		ContextMenuItem.ItemBuilder builder = new ContextMenuItem.ItemBuilder();
+		builder.setListener(new ContextMenuAdapter.ItemClickListener() {
+			@Override
+			public boolean onContextMenuClick(ArrayAdapter<ContextMenuItem> adapter, int itemId, int position, boolean isChecked) {
+				if (isChecked)
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/activities/MapActivityLayers.java]: We put braces for all branches

File: OsmAnd/src/net/osmand/plus/dialogs/ConfigureMapMenu.java [hunk 2]
```diff
@@ -125,17 +126,18 @@ public boolean onRowItemClick(ArrayAdapter<ContextMenuItem> adapter, View view,
...
 				item.setColorRes(isChecked ? R.color.osmand_orange : ContextMenuItem.INVALID_ID);
 				adapter.notifyDataSetChanged();
 			}
 			if (itemId == R.string.layer_poi) {
-				settings.SELECTED_POI_FILTER_FOR_MAP.set(null);
+				pfh.clearSelectedPoiFilters();
 				if (isChecked) {
 					selectPOILayer(adapter, adapter.getItem(pos));
 				} else {
-					adapter.getItem(pos).setDescription(POIMapLayer.getSelectedPoiName(ma.getMyApplication()));
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/dialogs/ConfigureMapMenu.java]: Doesn't look correct change.

File: OsmAnd/src/net/osmand/plus/OsmandSettings.java [hunk 3]
```diff
@@ -2319,9 +2322,21 @@ public boolean setLastSearchedIntersectedStreet(String street, LatLon l) {
...
 	}
 
-
+	// Avoid using this property, probably you need to use PoiFiltersHelper.getSelectedPoiFilters()
 	public final OsmandPreference<String> SELECTED_POI_FILTER_FOR_MAP = new StringPreference("selected_poi_filter_for_map", null).makeGlobal().cache();
 
+	public Set<String> getSelectedPoiFilters() {
+		Set<String> result = new LinkedHashSet<>();
+		String filtersId = SELECTED_POI_FILTER_FOR_MAP.get();
+		if (filtersId != null)
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/OsmandSettings.java]: Codestyle

File: OsmAnd/src/net/osmand/plus/helpers/WaypointDialogHelper.java [hunk 4]
```diff
@@ -603,8 +603,7 @@ protected View createItemForRadiusProximity(final FragmentActivity ctx, final in
 		if (type == WaypointHelper.POI) {
 			v = ctx.getLayoutInflater().inflate(R.layout.drawer_list_radius_ex, null);
 			AndroidUtils.setTextPrimaryColor(mapActivity, (TextView) v.findViewById(R.id.titleEx), nightMode);
-			String descEx = waypointHelper.getPoiFilter() == null ? ctx.getString(R.string.poi) : waypointHelper
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/helpers/WaypointDialogHelper.java]: The same as above

**Pull request merged**